src/leveled_sst.erl

```diff
@@ -2669,69 +2682,115 @@ binaryslot_trimmed(
-                MidBlockList,
-                get_righthand_blocks(B4, B5, PressMethod, EndKey)
+                in_range(
+                    get_lefthand_blocks(
+                        B1, B2, BlockMethod, StartKey, EndKey),
+                    StartKey,
+                    all
+                ),
+                MidBlockFetchFun(all),
+                in_range(
+                    get_righthand_blocks(
+                        B4, B5, BlockMethod, StartKey, EndKey),
+                    all,
+                    EndKey
+                )
             );
         lt_mid ->
             in_range(
-                get_lefthand_blocks(B1, B2, PressMethod, StartKey),
-                all,
-                EndKey);
+                get_lefthand_blocks(
+                    B1, B2, BlockMethod, StartKey, EndKey),
+                StartKey,
+                EndKey
+            );
         le_mid ->
-            append(
-                get_lefthand_blocks(B1, B2, PressMethod, StartKey),
-                in_range(MidBlockList, all, EndKey)
+            in_range(
+                append(
+                    get_lefthand_blocks(
+                        B1, B2, BlockMethod, StartKey, EndKey),
+                    MidBlockFetchFun(all)
+                ),
+                StartKey,
+                EndKey
             );
         mid_only ->
-            in_range(MidBlockList, StartKey, EndKey);
+            in_range(MidBlockFetchFun(all), StartKey, EndKey);
         ge_mid ->
-            append(
-                in_range(MidBlockList, StartKey, all),
-                get_righthand_blocks(B4, B5, PressMethod, EndKey)
+            in_range(
+                append(
+                    MidBlockFetchFun(all),
+                    get_righthand_blocks(
+                        B4, B5, BlockMethod, all, EndKey)
+                ),
+                StartKey,
+                EndKey
             );
         gt_mid ->
             in_range(
-                get_righthand_blocks(B4, B5, PressMethod, EndKey),
+                get_righthand_blocks(
+                    B4, B5, BlockMethod, StartKey, EndKey),
                 StartKey,
-                all)
+                EndKey
+            )
     end.
 
-get_lefthand_blocks(B1, B2, PressMethod, StartKey) ->
-    BlockList2 = deserialise_block(B2, PressMethod),
-    case previous_block_required(
-            fetchends_rawblock(BlockList2), StartKey) of
+get_lefthand_blocks(B1, B2, BlockMethod, StartKey, EndKey) ->
+    {Top, Tail, InnerLeftBlockFetchFun} =
+        leveled_sstblock:get_topandtail(B2, BlockMethod),
+    case previous_block_required({Top, Tail}, StartKey) of
         true ->
-            in_range(deserialise_block(B1, PressMethod), StartKey, all)
-            ++ BlockList2;
+            case this_leftblock_required({Top, Tail}, EndKey) of
+                true ->
+                    append(
+                        leveled_sstblock:get_all(B1, BlockMethod),
```

Comment:
The name `get_all` in that module does no longer refer to "deserialize", so one wonders whether the result is deserialized. When looking in the code for leveled_sstblock, it clearly is for version 0 blocks, but harder to distinguish for version 1 blocks (see comment in file).
I am not sure this fits the bill, but could one rename `get_all` to something making clear that what type of objects one gets returned (leveled_codec:ledger_kv() according to spec).
`get_all_deserialized`?

Reply:
I've tried to adjust the comments to the top to make this a bit easier to understand